ZPastel.Test/UserEndpointTests.cs

```diff
@@ -0,0 +1,180 @@
+
+            var client = GetClient();
+
+            var createResponse = await client.PostAsync("api/users/save", content);
+            createResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var createdUserContent = await createResponse.Content.ReadAsStringAsync();
+            var userResource = Newtonsoft.Json.JsonConvert.DeserializeObject<UserResource>(createdUserContent);
+
+            userResource.Should().NotBeNull();
+            userResource.CreatedById.Should().Be(command.CreatedById);
+            userResource.CreatedOn.Should().Be(command.CreatedOn);
+            userResource.Email.Should().Be(command.Email);
+            userResource.FirebaseId.Should().Be(command.FirebaseId);
+            userResource.Id.Should().BeGreaterThan(0);
+            userResource.LastModifiedById.Should().Be(command.LastModifiedById);
+            userResource.LastModifiedOn.Should().Be(command.LastModifiedOn);
+            userResource.Name.Should().Be(command.Name);
+            userResource.PhotoUrl.Should().Be(command.PhotoUrl);
+        }
+
+        [Fact]
+        public async Task SaveUser_WithUserInRepository_ShouldUpdateUser()
+        {
+            var client = GetClient();
+            var response = await client.GetAsync("api/users/user-1");
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var userContent = await response.Content.ReadAsStringAsync();
+            var user = Newtonsoft.Json.JsonConvert.DeserializeObject<UserResource>(userContent);
+
+            user.Email = "newemail@test.com";
+            user.LastModifiedById = 2;
+            var now = DateTime.Now;
+            user.LastModifiedOn = now;
+            user.Name = "New Name";
+            user.PhotoUrl = "www.newurl.com/photo.jpg";
+
+            var content = new StringContent(JsonSerializer.Serialize(user), Encoding.UTF8, "application/json");
+
+            var createResponse = await client.PostAsync("api/users/save", content);
+            createResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var createdUserContent = await createResponse.Content.ReadAsStringAsync();
+            var userResource = Newtonsoft.Json.JsonConvert.DeserializeObject<UserResource>(createdUserContent);
+
+            userResource.Should().NotBeNull();
+            userResource.CreatedById.Should().Be(1);
+            userResource.CreatedOn.Should().BeAfter(DateTime.MinValue);
+            userResource.Email.Should().Be("newemail@test.com");
+            userResource.FirebaseId.Should().Be("user-1");
+            userResource.Id.Should().Be(1);
+            userResource.LastModifiedById.Should().Be(2);
+            userResource.LastModifiedOn.Should().Be(now);
+            userResource.Name.Should().Be("New Name");
+            userResource.PhotoUrl.Should().Be("www.newurl.com/photo.jpg");
+        }
+
+        [Fact]
+        public async Task SaveUser_WithEmptyName_ShouldThrowArgumentException()
```

Comment:
Update the test name to ...ShouldReturnBadRequest

Reply:
Check in this test if the operation return which field is causing the problem

Reply:
Done.